arrow-fx-coroutines/src/test/kotlin/arrow/fx/coroutines/FlowTest.kt

```diff
@@ -0,0 +1,66 @@
+package arrow.fx.coroutines
+
+import io.kotest.assertions.throwables.shouldThrow
+import io.kotest.matchers.longs.shouldBeGreaterThanOrEqual
+import io.kotest.matchers.longs.shouldBeLessThan
+import io.kotest.matchers.shouldBe
+import io.kotest.property.Arb
+import io.kotest.property.arbitrary.int
+import io.kotest.property.arbitrary.positiveInts
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.reduce
+
+class FlowTest : ArrowFxSpec(spec = {
+
+  "Retry - flow fails" {
+    checkAll(Arb.int(), Arb.positiveInts(10)) { a, n ->
+      var counter = 0
+      val e = shouldThrow<RuntimeException> {
+        flow {
+          emit(a)
+          if (++counter <= 11) throw RuntimeException("Bang!")
+        }.retry(Schedule.recurs(n))
+          .collect()
+      }
+      e.message shouldBe "Bang!"
+    }
+  }
+
+  "Retry - flow succeeds" {
+    checkAll(Arb.int(), Arb.int(5, 10)) { a, n ->
+      var counter = 0
+      val sum = flow {
+        emit(a)
+        if (++counter <= 5) throw RuntimeException("Bang!")
+      }.retry(Schedule.recurs(n))
+        .reduce { acc, int -> acc + int }
+
+      sum shouldBe a * 6
+    }
+  }
+
+  // Note: this takes 50-60 sec to run
```

Comment:
We should use KotlinX Coroutines Test here: https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-test/kotlinx.coroutines.test/run-blocking-test.html

That should make this test instant, whilst it should also stabalise any flakyness that might occur by doing time measurements on the JVM.

See: https://github.com/arrow-kt/arrow-fx/blob/master/arrow-fx-coroutines/src/test/kotlin/arrow/fx/coroutines/BracketCaseTest.kt#L21

Reply:
Didn't know this existed. Today I learned :smile: